### ISSUE TITLE
Explore plain content editable element for terminal buffer instead of navigation mode

### ISSUE THREAD

**Issue by Tyriar** [microsoft/vscode#169853]
Prototype: https://github.com/xtermjs/xterm.js/pull/4340

**Comment by Black3me3** [comment]
> Prototype: [xtermjs/xterm.js#4340](https://github.com/xtermjs/xterm.js/pull/4340)

**Comment by meganrogge** [comment]
Shift tab appears to list things with oh-my-zsh. 
https://user-images.githubusercontent.com/29464607/211429089-1f02a5ee-0b10-4445-9cc8-2014878261b2.mov

**Comment by Tyriar** [comment]
@meganrogge oh you're right, it does go to the shell. In that case I'd special case that in a11y mode here:

https://github.com/microsoft/vscode/blob/b535afbd27f160661a5586b65b1333a95197073c/src/vs/workbench/contrib/terminal/browser/terminalInstance.ts#L1022-L1037

Tab focus mode can be toggled to use it otherwise

**Comment by meganrogge** [comment]
@jooyoungseo it would be great if you could test this in tomorrow's insider's

**Comment by rmwilliams2023** [comment]
@meganrogge I can confirm @jooyoungseo 's comments here, using latest release builds of JWAS and NVDA.

**Comment by meganrogge** [comment]
@jooyoungseo @rmwilliams2023 I will try on Windows, but all of the above is working for me on mac.

**Comment by meganrogge** [comment]
I see that on Windows Alt+f1 does not work - you should be able to run the command `Terminal: Show Accessibility Help` though. I will fix that, thanks for letting me know. 

I can reproduce on Windows not being able to navigate via cursor. I am surprised by this as it's a content editable, which works on mac.

**Comment by jooyoungseo** [comment]
@meganrogge What I meant by textarea-style element is the contenteditable area, in theory, is a text area where standard cursor navigation works via arrow keys. I recommended you turned off screen readers' invisible virtual buffer, such as NVDA's browse mode or JAWS' virtual cursor, when testing this area. Otherwise, screen readers' arrow navigation keys would interrupt the native contenteditable cursor navigation.

**Comment by jooyoungseo** [comment]
@meganrogge  -- The most accessible and screen-reader-friendly textarea view in VSCode worth testing is `workbench.action.output`.

You know, you can Ctrl+Shift+U can trigger `workbench.action.output.toggleOutput` which is a textarea having `role = "textbox"`. This view is very accessible to screen readers! I was expecting that the terminal a11y buffer would behave something like this.

**Comment by jooyoungseo** [comment]
@meganrogge For the same reason, when testing any text editor area with VoiceOver, the QuickNav needs to be turned off.

**Comment by meganrogge** [comment]
Okay thanks for the info, I'll look into this today

**Comment by meganrogge** [comment]
I went with a textarea like you suggested @jooyounseo. I did not see any perf issues despite testing with 1000+ lines on windows and macOS, but think that could be one issue with this approach. 

@Tyriar was telling me that Chromium has a new feature that lets one place a cursor at any position, so when we adopt that, we might revert to the `div` instead.

**Comment by steverep** [comment]
@meganrogge the issues I mentioned in #171833 are still unresolved as of today's build.  The only thing that seems to have changed is that I now get feedback while typing commands.  However, ALT+F1 and CTRL+Up/Down seem to do nothing, and command output is no longer spoken.  I'm on Windows 10 with NVDA.

**Comment by pawin35** [comment]
I have tested the VS code insider with the new accessibility support for terminal and have found out the followings:
- The accessibility buffer do not work with NVDA with focus mode set to on (Arrowing up and down on the buffer do not provide any speech output)
- When switching to browse mode, the text can be read. (However, this is not the effective mode to use with VS code)
- When typing command that return an output on screen (such as `echo "hello"`), the displayed result do not auto-read by screen reader
- Since we do not use ctrl+up and down arrow anymore, the hint to use ctrl+up and down arrow to navigate should be removed.

**Comment by steverep** [comment]
This issue seems far from resolved.  Please reopen.

**Comment by meganrogge** [comment]
As I indicated, this will be resolved in the next insider's build, which will be available tomorrow.

**Comment by jooyoungseo** [comment]
The up-to-date accessibility report for this terminal patch has been filed via #171914.
